internal/provider/data_source_http.go

```diff
@@ -181,14 +199,14 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 			return
 		}
 
-		tr.TLSClientConfig.RootCAs = caCertPool
+		clonedTr.TLSClientConfig.RootCAs = caCertPool
```

Comment:
Refer to above comment

```suggestion
		if clonedTr.TLSClientConfig == nil {
			clonedTr.TLSClientConfig = &tls.Config{}
		}
		clonedTr.TLSClientConfig.RootCAs = caCertPool
```